scripts/shared/judgeJSFilesVersion.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+'use strict';
+
+const minimatch = require('minimatch');
+const {es5Path} = require('./esPath');
+
+module.exports = function(jsFiles) {
+  const es5Files = [];
+  for (let index in jsFiles) {
+    const file = jsFiles[index];
+    for (let pathIndex in es5Path) {
+      const pattern = es5Path[pathIndex];
+      const isES5 = minimatch(file, pattern);
```

Comment:
Is this efficient? What's the full `yarn lint` time before and after the change?

Reply:
I also want to refactor it. But currently I don't think of other refactor way.
But it doesn't affect too much efficiency, because this script will only be used on `yarn linc` to distinguish `es5` and `es6` files: [here](https://github.com/facebook/react/pull/11794/files#diff-07716b084a551f663fd6110a4c6a2bddR16).

For `yarn lint`, because the `es5`, and `es6` pattern is known, we don't need to judge files' ecma version.

Here is the `yarn lint` run time after the change:
```
Before: 22.50s
After: 17.86s.
```
The reason for runtime decrease is because, before the change, the `filePattern` was `[.]` ([ref](https://github.com/facebook/react/blob/master/scripts/tasks/eslint.js#L11)), after the change we use specific es path.

Reply:
Maybe there’s something clever we can do to have ESLint do the filtering instead of us? I’m not sure but since Prettier can do this, maybe you can find a way with ESLint API too.